File: sorting/bubblesort.py
numbers = [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0]

def bubbleSort(array):
    arrayLen = len(array)

    while arrayLen > 0:
        for i in range(0, arrayLen-1):
            if array[i] > array[i+1]:
                array[i], array[i+1] = array[i+1], array[i]
        arrayLen -= 1

    return array

# print(bubbleSort(numbers))
# 
# def bubbleSortRecur(array):
#     arrayLen = len(array)
#     if arrayLen == 1:
#         return array
#     if array[0] > array[1]:
#         array[0], array[1] = array[1], array[0]
#         return bubbleSortRecur(array[1:arrayLen])
#
# print(bubbleSortRecur(numbers))

File: sorting/bubblesort.js
//an elementary sort
//simple basic sorting algorithms that you would do off the top of your head

const numbers = [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0];

function bubbleSort(array) {
  arrayLen = array.length
  while(arrayLen > 0) {
    for(let i=0; i<arrayLen-1; i++){
      if (array[i] > array[i+1]) {
        let temp = array[i];
        array[i] = array[i+1];
        array[i+1] = temp;
      }
    }
    arrayLen--;
  }
}

bubbleSort(numbers);
console.log(numbers);
